### PR TITLE
Print the ocserv's certificate hash and key ID

### PR DESCRIPTION
**Pull request by nmav**
Signed-off-by: Nikos Mavrogiannopoulos <nmav@gnutls.org>

### CODE REVIEW

File: applications/luci-ocserv/luasrc/model/cbi/ocserv/main.lua [hunk 1]
```diff
@@ -27,6 +27,28 @@ local e = s:taboption("general", Flag, "enable", translate("Enable server"))
 e.rmempty = false
 e.default = "1"
 
+local o = s:taboption("general", Value, "sha_hash", translate("Server's certificate SHA1 hash"))
+local fd = io.popen("/usr/bin/certtool -i --infile /etc/ocserv/server-cert.pem |/bin/grep -A 1 SHA1|/usr/bin/tail -1|/bin/sed 's/\t//g'|/usr/bin/tr 'a-z' 'A-Z'", "r")
```
Comment (jow-) [inline applications/luci-ocserv/luasrc/model/cbi/ocserv/main.lua]: That command is extremely inefficient.
Simply `popen()` "/usr/bin/certtool -i --infile ..." and do the parsing in the Lua read loop:

    if fd then
            local found = false
            while true do
                    local ln = fd:read("*l")
                    if not ln then
                            break
                    elseif ln:match("SHA%-?1 fingerprint:") then
                            found = true
                    elseif found then
                            local hash = ln:match("([a-f0-9]+)")
                            o.placeholder = hash and hash:upper()
                            break
                    end
            end

            fd:close()
     end

File: applications/luci-ocserv/luasrc/model/cbi/ocserv/main.lua [hunk 2]
```diff
@@ -27,6 +27,28 @@ local e = s:taboption("general", Flag, "enable", translate("Enable server"))
...
+	fd:close()
+
+	if ln then
+		o.placeholder = ln
+	end
+end
+
+local o = s:taboption("general", Value, "pkid", translate("Server's Public Key ID"))
+local fd = io.popen("/usr/bin/certtool -i --infile /etc/ocserv/server-cert.pem |/bin/grep -A 1 'Public Key ID'|/usr/bin/tail -1|/bin/sed 's/\t//g'|/usr/bin/tr 'a-z' 'A-Z'", "r")
+if fd then local ln
```
Comment (jow-) [inline applications/luci-ocserv/luasrc/model/cbi/ocserv/main.lua]: Same here:
    if fd then
            local found = false
            while true do
                    local ln = fd:read("*l")
                    if not ln then
                            break
                    elseif ln:match("Public Key I[Dd]:") then
                            found = true
                    elseif found then
                            local hash = ln:match("([a-f0-9]+)")
                            o.placeholder = hash and hash:upper()
                            break
                    end
            end

            fd:close()
    end

**Comment by jow-** [comment]
Can you address the issues I highlighted above? If this has been taken care of I'm going to merge this.

**Comment by nmav** [comment]
Thanks I'll update it. Do you have any suggestion on how to make this field selectable and read-only? Currently it is read-only but cannot be copied & pasted.

**Comment by jow-** [comment]
Try using a `DummyValue` instead of a `Value` field and assign to `o.default` instead of `o.placeholder`

**Comment by nmav** [comment]
Done, thanks.

**Comment by jow-** [comment]
Looks, good - thank you :)

**Pull request merged**